pkg/query/flamegraph_arrow.go

```diff
@@ -528,6 +528,60 @@ func (fb *flamegraphBuilder) mergeSymbolizedRows(
 			fb.intersectLabels(r, t, recordLabelIndex, sampleIndex, cr)
 		}
 
+		// Compare the existing row's metadata values with the one we're merging.
+		// If these values differ we need to set the row's metadata column to null.
+		{
+			a := t.mappingFile.indices.Value(int(r.MappingFileIndices.Value(locationIndex)))
```

Comment:
all of these need to be wrapped in `if fb.builderX.IsValid`, and then an additional internal check that checks if the one being merged is null, as if that's the case it should also be set null. The reason both have to be checked is that setting something null after the fact can have the effect that the equality match might actually pass even if `.SetNull` was called, as `.SetNull` only updates the validity bitmap, not the value array.

Reply:
Good catch! Updating.